### DISCUSSION TITLE
Is there any possibility to speed up the tts and tts-server process?

### DISCUSSION THREAD

**Discussion by Approximetal** [coqui-ai/TTS#1534] [General Q&A]
Hi, thanks for making such a great project, it's very helpful for me to build a tts system. It's just that I find that when I run `tts` and `tts-server`, it takes quite a long time to wait, even with `tts -h` I have to wait 10-20 seconds (as the figure blow, it took 3 minutes to synthesize a short text) . I'm not sure where this time is spent, maybe on importing the environment or loading the model, but I was wondering if there is any possibility to speed up the process.

![image](https://user-images.githubusercontent.com/25227045/165017177-595cc4ad-f60e-44eb-8ca9-b33bc1930553.png)

**Comment by domcross** [comment]
Looks like you are running on CPU. You can try --cuda option to run on GPU, this speeds up inference.
In case you don't have GPU available try a VITS-model, these can achieve real-time factor of 1x or below on CPU.

**Comment by p0p4k** [comment]
Another tip to add on top of @domcross's reply is that you can pre-load the synthesizer and then use it for inference. 
https://github.com/coqui-ai/TTS/blob/c410bc58ef3bd07b72ab05d29bbdc2a6df47afea/TTS/bin/synthesize.py#L264-L274

**Reply by curious-broccoli** [reply]
What do you mean by pre-loading the synthesizer to reduce loading time? Instead of running the `tts` command every time I should run the synthesize.py file and only construct and import the Synthesizer once?

Just the importing of the Synthesizer takes about 9 seconds (CPU).

**Reply by p0p4k** [reply]
Yes, create a synthesizer object once. Then just call the `tts` function with appropriate parameters (text, speaker name, etc.). I used GPU with VITS model, works really fast.
Other than that in #1686 , you can see in inquire.py file (on L282), how I preload a synthesizer and later call it (on L183). That way even in CLI, TTS does not require you to re-load models which speeds up the inference for the following sentences.

**Comment by king-dahmanus** [comment]
is it included by default now? I mean if I update my git repository will it automatically preload the models? If not, then I suggest adding another arg to the tts cli, which is "--init", like if I want to synthesize using one model for a while, I can just do:
tts --init model_name=tts_models/en/ljspeech/vits
then it will load that model and then when I do
tts --text "something"
without a model name arg, it will use the preinitialized model by default. I think this will help for sure

**Comment by king-dahmanus** [comment]
uh, ignore the = please

**Reply by couldbejake** [reply]
How long did it take you?

**Comment by chuks** [comment]
Anyone tried running this on AWS?
